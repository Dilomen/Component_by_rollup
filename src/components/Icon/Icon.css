@font-face {font-family: "iconfont";
    src: url('//at.alicdn.com/t/font_1822642_8q45gg6uni2.eot?t=1589594666932'); /* IE9 */
    src: url('//at.alicdn.com/t/font_1822642_8q45gg6uni2.eot?t=1589594666932#iefix') format('embedded-opentype'), /* IE6-IE8 */
    url('data:application/x-font-woff2;charset=utf-8;base64,d09GMgABAAAAABNAAAsAAAAAMHQAABLwAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHEIGVgCJOgrGLLlfATYCJAOBZAt0AAQgBYRtB4VtG+goVQQ2DgDS+LtH9v91AjdF5LGouyCJJmW3rrnCasxRR0nQUecLKSE48NCNNos9ylFCeRMT90RMRExoMx8W93f8z5weLqLZUEoevl/+PG+X2n+p3TgGIxgfZTI+xkb6Czz8d99uM2/+slOYUSehQDaSLMIEA+wEsmI/z2/zz3uX93iCCohN5AqdYiCIBVjfmskKp88YsnZRxoxc47La7f+v/nSNP10lATDvOzWpZdkpGoYEkpyk4+IBbuwf6de+Y0GHlI4ADO8OxDkkZO8Z7WZsfX+mNrc6bAyL6+i323vajuFzp4RDRAiXCPj/53qzAwWkFG9ZGAJbVeUCb+Alb+B9zqRE+YwD6ZykgOAQU8JMy7vsAIUElm4lGbNynVlh1aqbGmV/krZliBPmw9j89nAVen60xwwUrFrU1YtJuLExlba79DygBPVOCel8KCsCypcsIlonjb8KeGRYXaFo0gW/g7/mD59+1qkhUzL1np/fiaVgiOqeHMScbF82QPJkLB9FjinOyLrABSlkO2sq1+0/rvAAPjzJVa9/cSZev/zRwXIcQh85jpW/a5zOqJ71ffWy3cFsl+pYEUtVk7UvrW+e4Wil3CTyP+QlWSgqtUar0xuMJrPFsxev3rz7UFBUUlZRVVPX0NTS1tHV0zcwNDI2MTUzt7C0sraxtXNwtCdVSMk7yG7IPhLVOpCIWUBG1AA5UQucEXXAOVEPXBANwCXRCFwRTcA10QzcEC3ALdEK3BGzgXtiDvBAzAUeiXnAEzEfeCYWAC/EQuCVsANvRBvwTiwCPojFQIFYAhSJpUCJWAaUieVAhVgBVImVQI1YBdSJ1UCD2AE0iRtAi7gJtIl2oEM8BLrEI6BH/Aj0+WECA34oagjiAmoEwgI1BtGCmoDoQU1BUkHNQNJAzUHSQS1AMkAtQQaBWoFkglqDZIHagGSD2oLkgNqBrICoHGQlgCO/OVXswYk+5A/w382Th7+aku4mlWtSUXhI4UrBSuneOisJBWdFKZo73ZywnGx+PghJ0GNaQzyAbC5A5bua3VxVGVaajqMenQXNYswcjFXu0EUuOqOvYlTxmqWewpFmV151uTfGrESGafbgmJXfW5T18qwvqTIraxgQe/3MR6jGV08y63yCDF6ba9YwutlW33oQo18+z7KurJJuhWvVJmY97pUjrwmFkSqdj6iboNbqinaiwMJUOSgQNpGh26BVPopwHJM09ZJklzG1OO+DomgnRwhTJACzCaY/JsMjhobcoZ7mXxk/KPENaO/ZsIMXgjMECatI2oZ2i9VOLOTOlBoK5AxtBSFby9wATaD3Y91qCgtIO6NdKArD+i8QFkB8yQYozUZDg19TvUINW1R1Lpt0GzXfBEn1SxbeBK/uXQLZ3OHwx/B9d3au34lHzO14h3s37cmvZBFrp5ytVAnQ9DJgssKrmTd6WFMbvANWbTlPSMNoRpEYu1M7G8QGaH9KnjjJER1sSHmzpW/Dxa+nS3uZkF2relxZy1JxRz5r/EMsb8QbLnoVLHkzQzkrtXIqiXHtyMpMktVPW0JzvYl052pHOQ8SLtYbswqLpglrjhLFx6Tc+tax5bYbbgr+lHAeQgEWZRoBFg8H7Kxh3I5rzGWhpjz7KYqomExxEHT2JXArp55BMFkQv/Wf9/bgz6Njhi9Fi6KAvknGJNy7urZj6nAd09I3tUKSb9ooxyRGRCWFXl4DQaiy9yKW5jyd+TBdWJQ5EjV0SWYhOiDonHyNekvx9RgJdvfpft7F0/sdJ1njqFMWfrCEKnnzEoxEvX9PX02j6oRlfcKW2GuTdtYZi3bbjlfb8qTr3V9MpLTt6+FEV2vXezuTyNOu+9Vz9T66l3XqRs4No4NujRuHr02gE25MGYIvjPB67ONde9KLI8nQuxMnejfHk/HXJw/2zg/HW6oXIreCTnhAzPz//8i2d/7vz4b9m4IRIYbziLFpnr3xpcfhrxOvf+5+4JNfM3/S+kESq0DcX4bu6SOLQpC3fUhok3NLGoMBGz0re8g3aHJV6ZQu4JjBGVDSg29XHXbAYvU9ZtLwQrdVO1QFVpNY3Dq9aRzkbgSAv/ra2rqAqpIoUHlb153CpRzjU+XAODco5T9mgOuoqlJjxIiZRG9FNj2Leh2N579M+x5L8I0H/GF4n937zgwEKCrED5YJNSzmKoug0ciujiYefeO0DFFP3/bC7n77wULuLCEtrKvlinzW11T1WSLw7nchHeJ2v99T/mSACqwTj9hjT0NXtdttJLMaIWdPKd6Yjvw0F5LsB3vauY/FpeSBTel3BkjiBvI3yxUR29c6WniuFW2Vbmy98Rudh0JMXtL4xNSjh2jMOIP4FyF/MqSidzhB1951y6K1s6iSe/Mp5F0u9tRvuO/nYqNo66DOErBB1NlumZ2xzOIG3WGEtuyEM3IvYP5+yEfAcUh5MkENY16UuZE7+H78zbQHNUK3Qyb+cfecP5hkryVQUHDyxRl5FjCr0+CznWl9oLT75jgKn0IyL2RH+EWQjI8gLzhhWFeXC8B7CAbj9FQt/EOVV9365HQ13so1itY3xCt14l4nIH9Kq+x7MECYv639lb33+5/0jU/poqcfdoXKq/GItSEpW1sGVdPmpky3VfOdd2TaJFj0XuFtQbLfclMQHnZNW5upjxjSMplYcJeStNpN0UZt36Vsbe3n1GyeuAQ0bcgZnCfjfXJZEThUacOQNvGMdUIZZpJlFiYSACZinJC7npYAfpxyvOF3ygnLVA7UbkbaQQXXIYqRgrIwpcUNg3iGBLkYGU8FYSHlFGu3YBdDTxV4R3iWnRHCZU2xqGFQueDyX58URUtVMAsZch22PEaSERAvQp3LgkdraYP8fsvFFPKEjYKoEYuf2QjTqc3SnamdljNRkHxJePo1DeN71/6wjTVotI5GiTUSMRpTb3xKHMxds6YrqXh1bzChU1TuWSPZ065Qas9LzzK+fmWcFf369cfjlmM/DV6L2tsRWq9tb/egHOwRUpSpeJoGNpUzEa9u1V48xCxn7LW7FdujFLk7pcgQv8haiRrt6q9USWpQ0ySXpGk5GK57d1BOybKcaTIRzL3ZZptn601uMDG5kFEgevCvFuOS0rY1i/xa5cF3HpRHHlal9lieUuX4+nKaVJM03cznNiRNXrGiamwLIiJ9cRYV54a+nrqq389TPlzvc/kw5T27S7nM2tfbZoi4ycDQkkZDY0ejmCr7zSksPycf8/1p1ZeZCm/moT4c5k8XQpPY4sbmOf15Vrl2ErCq+d1BY0d6TfNfLqPWnq618o6jbm6Brt0JnoVA+sjHgoEYIv90zxOigd/w/FTelo9+ewtmDJ1RUNj1ed7nm87//rkwX8M+lTqXAUeIxtBKfcoucynsR3+/ZO4n6Wj6rcAk9/RmFhRMi2z4+OGdLkmbvRFotLP78U60Cz1+jHZCQca6BG1uQrEm4eSu0yFT6tVThg5tOVE/RT1kqI/1a53+/e8GRZ1n341Iw9gGhI4Pzpqip7/nO2PQ9168UpZZWbU5wYJbchRZsoqq06Is1xLuO3LcCs+VnitelPqG48M3xVdV/FlQ5B4/WZTQFJGGDxziNXXy/Pna/uiGDEyPaLaNatamAp7WoisrE5iECSt0zRCoQMYSwuAyhXEH9guMbb7t/uLiv0wCu1HwubiYsQ53PEtdDESJcZSm9jo+Cz/+5uNxxnH05k3eFKutvZVeaaDJByS9SBAisAqrlaT9s1PFSeLx40TJktRNYrf4uPHJolTJpgBL4mRxarY/ucX1+wx10/SQEQUh05v9ifzfag1uGjt2mHrkHUEAkaCvLimu0icShyhmLUnXQHmaEXryAhMSAnPXQWS7u1Gbrk/Pz9lqHID7p2VgVmV9tvBKHifP7/c2NMgt0w1v+92Pk5d3JVtYr7TyYvzTcJ1pmlEXpY8yaqeZIjH/tBheWQJnbhZ3WpZVOnfmOso+HmyAaeHWiKCIISlxZH2+tCf6fAGPXrKFRw/aH9MzQlZPphqGaMLsqexZridm/whu0ebFuHfB+RhHvrTeOzXOrSQVKy7G0pv6cPC4BdBXTr7B8Ou6FjAY80H/eNX8AMNIPXnSyPdt5qGq2FqxJb62TkNM7eGqqr1iqqSKev/jcP7rJd8E0EwqFBqspUb/ROGyBIFJYC0VGoTxqaNDrP2yk9JiDKPT9Ut9l+pHpccaEtP6Z4eMHJM6mfWg2+F6LCrQZs4dE1BhttntjboOodsRk1xy2ZhvnL+Urq5WR5MGu91AkzqStrnz3Lv1xGykygubxmxlTsUZvPXrNJqqSh7TsuDRY7PesZVdCAJt7VxNOTN2Gr9tLElraLLUjed2idyslGR/mT8htIFI10teqk8nGkKvby2jQPvQsnHj6adInZe52J9TAza43s9ub23dYoN+nUktLcklF7OyhMLhu2hmRf/OvZOgzYP7/6vLFq5k6Nk3uMHL8QGbyZLGVw6KuqOYiYOtGiGRGUXoFDPbdqTQEezWHkHdxef0DE2HnQcELeIGKW9eZ9zaimQ8k6tv/e44fmejxLPPpUmln80EJTbAeIOXFl3Y3KLRemj51qW025ZmsrZAkSBcTFzStgL0x71OZpFqMuskTV7SnN9gNgfXOUvjkHycRbIcxe1WW2R1FEvHiYeIQDeKtHvSbbEPwA322+BsDcmFLxUbDtqX/8YmF745aOFPPfDv26GReGncCP2UGiZYNHpRgFzV/pHfS1SIYk1B5Xe3miPqwvFY/dMc5n4lbl1/vrGOxc9exMIvBQizhSradGMsx0+W0znW3UWa0zWGwxO8Xy8pscUUcdjL/saoa/oOB8dfaoVlRn5AWzafjXnkwrgrkpDExOrMEVvvlj+INYqIit5Xx2OVPKAjQUlAtl5PXcP+XsbmFMWU2CTr3wt41c4cGcWhA0v9OZYuk9rMAVp8cNmUCTL0XY4vqzKN/nD7bT/a4ey0ifmgx66MneTGexAVw/wU67OuKc2LdOeF0mdWsjr1O/NWyYcqqkqtTsDTK+Y8mlARdJ6ZyFJlEWZzlq9plnnj58zJq2iL80iTBw5M93JHLphdj5DksZOSdEDmEVfzhSOb9VLlzHo/nkihUFu4PoGq8G3FpcJRujSDyivzlZxHZaD0pcKTYni01TGejDQD7YH59PM94j2Fzzt+aRqVJGjWhZKoNEILWlOtP01ZGOc+EIlfSuKsXu1y9dNqTlKgKW/4atHul6MVOgirGTGmOiKLmdiBdWKJzCwvazfbvhPr0Pj+Mw/1f6gYh7RyT/yiwb2rk4rXdL1jadyu16x9SUQfAhrJARn5B9+9x2lEKvbweuOpxjjr9S39Tor4chLAQmPS1LWL3ul4OkuSrH4z0hh7I9h9vlY07tYvv/CRJBq/nZah2tpuJhZA161CeIfEEhYTOMZsHhNYYbbZTSo2FmLCLJJqHOXmntaE5wqdWbLhq5NTljU8Yq/pulym65Fcmohiw6RUoMap5qsFzgtKNQA4z5v6OP8fCr1/cwtSZn3Vjhh5SReDej/wAFJn3fAE0sFsF/oBAACdwq/Nel92Corp3etixMtyogUI/IodcXm2Ksd7EV/T/tQusj+ThLIUO34D9S+4lbZ2BG1Y+LcU/EfMiGbhtW1X4Zek2LL/IdoO/1v35XPLhv4L5IawlXRvKuREv6PYjNNR8MvzMSpzgfN8QHsqFOBhMzGA//BAnDrGwa2mRDJ2f190ocZ9I2EdsFUcAFjaHACnMscD/mgXnxc3TzHlQ0sGByaIGAQUKO2M+oEMAWwIZ0igII5hQQAkP54NfJgECDCGCwCooGnGYMCF/QwO7nCeQcCFu3ZG/d8ZArzhX4YELoYYFgzG+BbZ0BcT0iReYBXeXJ+GHKfgNJ1YpT/Q1GjF6zZ7/UAKLTaHIPV57gUJIuM+5dYcVYMJwp05z14OMbLJwr9wum1V82kXbA16i63j7mUSL7AKb65POjo5TmH7OmG+/w80NVqJiLof/QdS6OTNYbvnsF+kxBW1LBXLrTkqkYPRawp35kxkRC2LTQ6+2S+cbtsUaj7tyPICr7TN39dNSwlh8MVU/ytOGZLIFCqNzmCy2Bwu7/nl9e39o1AslSvVWr3RbLU73V5/MByNJ9PZfLFcrTfb3f5wPIXrTdfC/W0ciYuYucgFntwppcDzHhFBu3u+p1KzjlItF9fJEc3ts87yv6IocYLFQk0r6XZDpQ5cdVwzU2Nxt5IoNeuPr1eeyfUEnocdSrENRpnvEA5hlAmutaLDAtX4ld/xxCiyszl5p/hgpJvyX61ggEdm0V6xN4xszkVZaooi38/QcVLhOHXsEekqVp4DD7UUK4h4EJDraje0q9KL9IdBTQYMHD1kWmLDGWmiVLXe94qzqd8AYq9N57/V3GtPqS9K6pfIsO6S424CxbhqfnQW3gcCDOuOQLuIvkQJwjtUQGDR5ugY0S8I4T0qoPSgwIprKyZEh1dSQNILAAA=') format('woff2'),
    url('//at.alicdn.com/t/font_1822642_8q45gg6uni2.woff?t=1589594666932') format('woff'),
    url('//at.alicdn.com/t/font_1822642_8q45gg6uni2.ttf?t=1589594666932') format('truetype'), /* chrome, firefox, opera, Safari, Android, iOS 4.2+ */
    url('//at.alicdn.com/t/font_1822642_8q45gg6uni2.svg?t=1589594666932#iconfont') format('svg'); /* iOS 4.1- */
  }
  
  .iconfont {
    font-family: "iconfont" !important;
    font-size: 16px;
    font-style: normal;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
  
  .icon-check-circle:before {
    content: "\e77d";
  }
  
  .icon-close-circle:before {
    content: "\e77e";
  }
  
  .icon-info-circle:before {
    content: "\e77f";
  }
  
  .icon-left-circle:before {
    content: "\e780";
  }
  
  .icon-down-circle:before {
    content: "\e781";
  }
  
  .icon-minus-circle:before {
    content: "\e782";
  }
  
  .icon-plus-circle:before {
    content: "\e783";
  }
  
  .icon-play-circle:before {
    content: "\e784";
  }
  
  .icon-question-circle:before {
    content: "\e785";
  }
  
  .icon-right-circle:before {
    content: "\e786";
  }
  
  .icon-timeout:before {
    content: "\e787";
  }
  
  .icon-up-circle:before {
    content: "\e788";
  }
  
  .icon-warning-circle:before {
    content: "\e789";
  }
  
  .icon-sync:before {
    content: "\e78a";
  }
  
  .icon-redo:before {
    content: "\e78b";
  }
  
  .icon-message:before {
    content: "\e78c";
  }
  
  .icon-poweroff:before {
    content: "\e78d";
  }
  
  .icon-piechart:before {
    content: "\e78e";
  }
  
  .icon-setting:before {
    content: "\e78f";
  }
  
  .icon-eye:before {
    content: "\e790";
  }
  
  .icon-location:before {
    content: "\e791";
  }
  
  .icon-edit-square:before {
    content: "\e792";
  }
  
  .icon-export:before {
    content: "\e793";
  }
  
  .icon-save:before {
    content: "\e794";
  }
  
  .icon-appstore:before {
    content: "\e795";
  }
  
  .icon-close-square:before {
    content: "\e796";
  }
  
  .icon-control:before {
    content: "\e797";
  }
  
  .icon-codelibrary:before {
    content: "\e798";
  }
  
  .icon-detail:before {
    content: "\e799";
  }
  
  .icon-minus-square:before {
    content: "\e79a";
  }
  
  .icon-plus-square:before {
    content: "\e79b";
  }
  
  .icon-check-square:before {
    content: "\e7a8";
  }
  
  .icon-like:before {
    content: "\e7c8";
  }
  
  .icon-unlike:before {
    content: "\e7c9";
  }
  
  .icon-folder:before {
    content: "\e7d1";
  }
  
  .icon-folder-open:before {
    content: "\e7d2";
  }
  
  .icon-folder-add:before {
    content: "\e7d3";
  }
  
  .icon-scan:before {
    content: "\e7d4";
  }
  
  .icon-right:before {
    content: "\e807";
  }
  
  .icon-left:before {
    content: "\e808";
  }
  
  .icon-up:before {
    content: "\e809";
  }
  
  .icon-down:before {
    content: "\e80a";
  }
  
  .icon-check:before {
    content: "\e837";
  }
  
  .icon-close:before {
    content: "\e83a";
  }
  
  .icon-check-circle-fill:before {
    content: "\e84a";
  }
  
  .icon-left-circle-fill:before {
    content: "\e84b";
  }
  
  .icon-down-circle-fill:before {
    content: "\e84c";
  }
  
  .icon-minus-circle-fill:before {
    content: "\e84d";
  }
  
  .icon-close-circle-fill:before {
    content: "\e84e";
  }
  
  .icon-info-circle-fill:before {
    content: "\e84f";
  }
  
  .icon-up-circle-fill:before {
    content: "\e850";
  }
  
  .icon-right-circle-fill:before {
    content: "\e851";
  }
  
  .icon-plus-circle-fill:before {
    content: "\e852";
  }
  
  .icon-search:before {
    content: "\e952";
  }
  
  .icon-folder-fill:before {
    content: "\e898";
  }
  
  .icon-folder-open-fill:before {
    content: "\e899";
  }
  