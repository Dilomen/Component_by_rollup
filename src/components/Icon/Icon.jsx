import React, { Component, isValidElement } from "react";
import { ReactSVG } from "react-svg";
import "./Icon.css";
class Icon extends Component {
  renderIcon = () => {
    const { icon, className = "" } = this.props;
    if (typeof icon === "string" && icon.slice(0, 3) !== "<svg") {
      return <i className={`iconfont icon-${icon} ${className}`}></i>;
    }
    if (isValidElement(icon) || icon instanceof Element) {
      return <i className={className}>{icon}</i>;
    }
    return <ReactSVG src={icon} className={className}></ReactSVG>;
  };
  render() {
    const { icon } = this.props;
    if (!icon) return null;
    return <>{this.renderIcon()}</>;
  }
}

export default Icon;
