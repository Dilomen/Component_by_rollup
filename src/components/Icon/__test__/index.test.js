import React from "react";
import Icon from "../index";
import { renderSnapshot, mountComponent } from "@tests";

mountComponent(
  <Icon icon="search" />
);

renderSnapshot(
  <Icon icon="search" />
);
