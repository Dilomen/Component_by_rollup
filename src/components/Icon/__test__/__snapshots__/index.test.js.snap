// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component could be rendered snapshots without conflict render Snapshots 1`] = `
<i
  className="iconfont icon-search "
/>
`;
